src/registrar/assets/js/get-gov-admin.js

```diff
@@ -63,6 +63,25 @@ function openInNewTab(el, removeAttribute = false){
     checkToListThenInitWidget('id_alternative_domains_to', 0);
 })();
 
+/** An IIFE to capitalize statuses in the Domain Application status dropdown
+*/
+(function (){
```

Comment:
The JS approach seems like a bandaid instead of a cleaner structural revision. I say that because your approach to Domain was so much more streamlined.